src/java/org/apache/cassandra/service/accord/async/AsyncOperation.java

```diff
@@ -311,12 +315,28 @@ public void run()
         }
     }
 
+    private boolean preRun()
+    {
+        commandStore.checkInStoreThread();
+        try
+        {
+            return runInternal(true);
+        }
+        catch (Throwable t)
+        {
+            logger.error(String.format("Operation %s failed", this), t);
+            fail(t);
+            return false;
```

Comment:
if we fail we will rerun (aka schedule in the background)?

Reply:
We can't run if we've failed the task?

Reply:
```
if (!commandStore.inStore() || preRun())
            commandStore.executor().execute(this);
```

we are just a simple `Runnable`, so we will call the `run` method... we don't check if we are completed atm

Reply:
But we return `false` so we won't submit ourselves - and even if we did, we've set our internal status to FAILED so the `run` method will do nothing.